src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,72 @@
+import { Link, useSearchParams } from 'react-router-dom';
+import classnames from 'classnames';
+import { getSearchWith, SearchParams } from '../utils/searchHelper';
+import { Filter } from '../types/Filter';
+import { SearchLink } from './SearchLink';
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const sexSearch = searchParams.get('sex');
+
+  function setSearchWith(params: SearchParams) {
+    const search = getSearchWith(searchParams, params);
+
+    setSearchParams(search);
+  }
+
+  function getCenturies(min: number, max: number) {
+    const centuryNumbers = [];
+
+    for (let i = min; i <= max; i++) {
+      centuryNumbers.push(i);
+    }
+
+    return centuryNumbers;
+  }
+
+  function handleQueryChange(e: React.ChangeEvent<HTMLInputElement>) {
+    setSearchWith({ query: e.target.value || null });
+  }
+
+  const getLinkStyle = (item: Filter) => {
+    if (
+      (item === Filter.All && !sexSearch) ||
+      (item === Filter.Male && sexSearch === 'm') ||
+      (item === Filter.Female && sexSearch === 'f')
+    ) {
+      return 'is-active';
+    }
+
+    return '';
+  };
+
+  const getSexFilter = (sexFilter: Filter) => {
+    switch (sexFilter) {
+      case 'Female':
+        return 'f';
```

Comment:
Replace magic strings like `f`, `m` with `enums `for better maintainability and type safety.